docs/dnspolicy/managed-zone.md

```diff
@@ -38,32 +82,38 @@ There are plans to make this more customizable and dynamic in the future, [work
+  domainName: mydomain.example.com
+  description: "My Managed Zone"
   dnsProviderSecretRef:
-    Name: my-credential
-    NameSpace: ns
+    name: my-aws-credentials
+    namespace: multicluster-gateway-controller-system
 ```
 
-### Secret Ref
-
-This is a reference to a secret that contains a credential for accessing the DNS Provider. See [DNSProvider](dns-provider.md) for more details.
-
-### Optional fields
-The following fields are optional:
-#### ID
-By setting the ID, you are referring to an existing zone in the DNS provider which MGC will use to manage the DNS of this zone.
-By leaving the ID empty, MGC will create a zone in the DNS provider, and store the reference in this field.
-
-#### Description
-This is simply a human-readable description of this resource (e.g. "Use this zone for the staging environment")
-
-#### ParentManagedZone
-This allows a zone to be owned by another zone (e.g test.api.domain.com could be owned by api.domain.com), MGC will use this owner relationship to manage the NS values for the subdomain in the parent domain.
-Note that for this to work, both the owned and owner zones must be in the Route53 account accessible by MGC.
+#### Spec fields
+| Key                    | Example Value                                                             | Required? | Description                                |
+|------------------------|---------------------------------------------------------------------------|-----------|--------------------------------------------|
+| `dnsProviderSecretRef` | `name: my-credential, namespace: multicluster-gateway-controller-system ` | Required  | Ref to DNS Provider Secret                 |
+| `domainName`           | `myapps.example.com`                                                      | Required  | Root Domain Name for this ManagedZone      |
+| `id`                   | `Z0WDADW1234`                                                             | Optional  | Zone ID for an existing Zone in GCP or AWS |
+
+#### Additional notes on spec fields
+
+* `dnsProviderSecretRef` 
+  * This is a reference to a `Secret` that contains a credential for accessing the DNS Provider. See [DNSProvider](dns-provider.md) for more details.
+* `id`
+  * By setting the `id`, you are referring to an existing zone in the DNS provider, which MGC will use to manage the DNS of this zone.
+  * By leaving the `id` empty, MGC will create a zone in the DNS provider, and store the reference in this field.
+* `description`
+  * This is simply a human-readable label/description of this resource (e.g. "Use this zone for the staging environment").
+* `ParentManagedZone`
+  * This allows a zone to be owned by another zone (e.g test.api.domain.com could be owned by api.domain.com), MGC will use this owner relationship to manage the NS values for the subdomain in the parent domain.  
+  * Note that for this to work, both the owned and owner zones must be in the Route53 account accessible by MGC.
```

Comment:
Is it possible to generate docs directly from CRDs? Would be nice if we could make them self documenting and have a reference section in the docs for each of our resources (ManagedZone, TLSPolicy, DNSPolicy etc..) and wherever we refer to them link to those docs.

Reply:
API Reference is what I'm thinking https://cert-manager.io/docs/reference/api-docs/. seems to use https://github.com/ahmetb/gen-crd-api-reference-docs

Think this would look better than the various ways we are currently trying to describe the APIs in different docs.

@jasonmadigan @maleck13 wdyt?

Reply:
> Is it possible to generate docs directly from CRDs? Would be nice if we could make them self documenting and have a reference section in the docs for each of our resources (ManagedZone, TLSPolicy, DNSPolicy etc..) and wherever we refer to them link to those docs.

I was thinking the same thing while writing these (that there must be some tooling to generate MD tables or whatever for API docs like these).

Reply:
I like the format of cert-manager's docs - will create an issue to look at using that tool to generate them

Reply:
Created an issue: https://github.com/Kuadrant/multicluster-gateway-controller/issues/558